### PR TITLE
Add CITATION.cff file and fix arxiv links

### PR DESCRIPTION
**Pull request by Vindaar**
Hey,

not sure if you recognize this handle, but we've talked at a few IAXO collaboration meetings in the past (Sebastian from Bonn).

I just wanted to cite your repository for my thesis (independent of the paper) and noticed there isn't a CITATION file. Adding one means there's automatically a button on Github on the side bar (where the README, activity, stars etc buttons are) to get the citation in the format of choice.

Example here: https://github.com/SciNim/xrayAttenuation

Github docs here: https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-citation-files

I'm not sure if the data I put in there is accurate. I used `v0.8b` as the version, because that is the one mentioned in the paper (and hence your institution @sebhoof; I noticed on the newer paper you have two different institutions assigned).

While looking at the README I also noticed a bunch of arxiv links are broken (was that format ever allowed?). 

#### On an unrelated note:

I saw your latest preprint (about axion helioscopes as "solar thermometers"). While I've only had a _very quick_ glance at it, first of all it's an interesting concept (for the ambitious among us ;) ). In any case, you mention the uncertainty of the X-ray optics in the whole. From my side I would expect that to be almost the biggest trouble in all this (having just spent another few more weeks related to raytracing the LLNL telescope used at CAST). 

However, it seems doable to use a raytracer like https://github.com/Vindaar/TrAXer to do this. I imagine one could effectively do gradient descent to backpropagate from the image to a matching axion emission. Ideally, this could even be done using automatic differentiation. I remember some Zygote.jl example about inverse rendering; while googling for it I found something even more applicable in a sense 

- https://github.com/avik-pal/RayTracer.jl
- https://arxiv.org/abs/1907.07198

(I imagine applying that to the axion problem would be tricky due to the solar emission being a rather non standard thing in terms of raytracers, let alone X-ray reflectivities).

In my raytracing code I linked I suppose it should be feasible to do inverse rendering at least if one bases it on dual number autograd. To do source code transformation like Zyogte.jl to generate the backward pass is - uhm - more tricky obviously. Anyway, computing the numerical derivative is not a problem for sure though.

Just food for thought!

### CODE REVIEW

**Comment by sebhoof** [comment]
Hi Sebastian, many thanks for suggesting this! I actually hadn't noticed these cffs before you mentioned them, but it's a useful thing to have.

We had a long discussion about whether or not to create a DOI for this repo etc. There are two main reasons we didn't do it. First, writing software code typically and sadly doesn't help you getting a job in physics as much as citations do. So we wanted the original paper to be cited.

Second, similar to e.g. my [sebhoof/snax](https://github.com/sebhoof/snax) repo, we always had the idea of expanding (and eventually renaming) the repo. Now, while I'm generally in favour of making it "easy" for the user, we didn't want to make it too easy in this case. Depending on what parts of the code (and what solar models etc.) have been used, there are a number of works that should be acknowledged. Offering an "easy way out" will lead to some people using it and not reading the README, which would make us (and the people who worked on solar models and opacity computations) rather upset.

We actually did have some unfortunate experience with people not acknowledging our work and repo appropriately. You could argue that we could've helped them more but, at the end of the day, this is just poor scientific conduct and it was _their_ responsibility to cite properly. Some people might find this too narrow-minded, but I just refuse to accept the laziness of people who don't care to do their jobs properly. The broken arXiv links aside, I think we've provided reasonable help with the bib file and README. However, it's probably good to update and restructure the README to make this more obvious, which I'll do once we have the JCAP DOI for our recent preprint.

On another note, I'd almost never to add all the information the cff supports; institutions change and I don't consider it very relevant information, adding a version number is both annoying because it needs updating and a user might choose a different version than the current one for one reason or another (so the "easy way out" might lead again to improper citation).

So I won't be adding the cff file to this repo, but I'll add one to my [sebhoof/bibcom](https://github.com/sebhoof/bibcom) repo (which also has a citable DOI from Zenodo). Again, thanks for making me aware of this!

As for the broken arXiv links, I think this is a copy-paste error from when I updated the README. Thanks for spotting it; I'll cherry pick this commit to merge.

Re the X-ray optics: I wanted to look at ray-tracing for some other reason anyway, so I'll send you an email about it over the next couple of days!

Finally: If you only want to cite the repo in your thesis, other than just citing a URL or linking it in a footnote, and en lieu of a generally used citation format for software, you could use this BibTeX entry, which should be supported by most citation styles:
```
@misc{github_solaraxionflux,
	author = {Hoof, Sebastian and Thormaehlen, Lennert},
	title = {{SolarAxionFlux. A C++/Python library for calculating the solar axion flux for different solar models and opacity codes.}},
	year = {2021},
	note = {[Github repository]},
	howpublished = {Available on Github at \url{https://github.com/sebhoof/SolarAxionFlux}}
}
```

**Comment by sebhoof** [comment]
I merged 48f8e24 into the master branch now; thanks again!

**Comment by Vindaar** [comment]
Yeah, those are fair points. I can see arguments for "both sides" if you will. From a more software oriented point of view it's nice being able to just have a quick way to *cite the source code*. But I totally agree that if people use that as an excuse to not cite the other, related work it's very annoying (in my case in particular I specifically refer to your code as an alternative to Johanna's code here https://github.com/jovoy/AxionElectronLimit/blob/master/src/readOpacityFile.nim ).

At least in terms of renaming the repository, that should not be a big problem. If you rename or even move a git repository on Github the old name simply becomes a redirect. And regarding the institution, while I agree that it is generally not very important _in principle_ it is useful in the sense as it allows for easier disambiguation of people, as I know Mr. Foo Bar refers to _that_ person who was at institute Baz in the year XYZ when the citation was created (as an alternative some ID identifying the researcher would work even better of course, ORCID or whatever).

Thanks for the BibTeX entry, I'll use that instead. And sure, just send me a mail about the X-ray optics if you want!
